doc/releases/changelog-dev.md

```diff
@@ -55,6 +56,9 @@
   a sparse matrix.
   [(#6173)](https://github.com/PennyLaneAI/pennylane/pull/6173)
 
+* The `Hermitian` operator now has a `compute_sparse_matrix` implementation.
+  [(#6225)](https://github.com/PennyLaneAI/pennylane/pull/6225)
+
```

Comment:
I don't think this entry belongs under the "Capturing and representing hybrid programs" header.